### PR TITLE
Question entity

### PR DESCRIPTION
**Pull request by FedenkoM**

### CODE REVIEW

File: src/main/java/ita/softserve/course_evaluation/service/AnswerToFeedbackService.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import ita.softserve.course_evaluation.dto.AnswerDto;
+
+import java.util.List;
+
+public interface AnswerToFeedbackService {
+
+    List<AnswerDto> getAllAnswer();
+    AnswerDto saveAnswer(AnswerDto answer);
+    AnswerDto findAnswerById(long id);
+    void deleteAnswerById(long id);
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation/service/AnswerToFeedbackService.java:12]: please, add blank line here

File: src/main/java/ita/softserve/course_evaluation/service/QuestionService.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import ita.softserve.course_evaluation.dto.QuestionDto;
+import ita.softserve.course_evaluation.entity.Question;
+
+import java.util.List;
+
+public interface QuestionService {
+    List<QuestionDto> getAllQuestion();
+    Question saveQuestion(QuestionDto question);
+    QuestionDto findQuestionById(long id);
+    void deleteQuestionById(long id);
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation/service/QuestionService.java:13]: please, add blank line here

File: src/main/java/ita/softserve/course_evaluation/service/impl/AnswerToFeedbackServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        answerRepository.save(AnswerDtoMapper.fromDto(answer));
+        return answer;
+    }
+
+    @Override
+    public AnswerDto findAnswerById(long id) {
+        return AnswerDtoMapper
+                .toDto(answerRepository
+                        .findById(id)
+                        .orElseThrow(() -> new RuntimeException("Answer was not found for id: " + id)));
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation/service/impl/AnswerToFeedbackServiceImpl.java:37]: It's better to create a specific exception, like AnswerNotFoundException, but I believe if there is no answer in DB - we get ```null```

**Review by Forfend — CHANGES_REQUESTED** [review]

File: src/main/java/ita/softserve/course_evaluation/entity/Feedback.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.ManyToOne;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import java.time.LocalDateTime;
+import java.util.Objects;
+
+@Entity
+public class Feedback {
```
Comment (Forfend) [inline src/main/java/ita/softserve/course_evaluation/entity/Feedback.java:14]: ```@Table```

**Review by Forfend — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 5]
```diff
@@ -34,6 +34,24 @@
             <artifactId>postgresql</artifactId>
             <scope>runtime</scope>
         </dependency>
+        <!-- https://mvnrepository.com/artifact/io.springfox/springfox-boot-starter -->
+        <dependency>
+            <groupId>io.springfox</groupId>
+            <artifactId>springfox-boot-starter</artifactId>
+            <version>3.0.0</version>
+        </dependency>
+        <!-- https://mvnrepository.com/artifact/io.springfox/springfox-swagger-ui -->
```
[the remark this answers is not in the document]
Comment (FedenkoM) [reply pom.xml]: done

File: src/main/java/ita/softserve/course_evaluation/controller/AnswerToFeedbackController.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+    @GetMapping()
+    public ResponseEntity<List<AnswerDto>> getAllAnswers(){
+        final List<AnswerDto> answerList = answerService.getAllAnswer();
+        return answerList != null && !answerList.isEmpty()
+                ? new ResponseEntity<>(answerList, HttpStatus.OK)
+                : new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @PostMapping("/addAnswer")
```
[the remark this answers is not in the document]
Comment (FedenkoM) [reply src/main/java/ita/softserve/course_evaluation/controller/AnswerToFeedbackController.java]: edite annotation

File: src/main/java/ita/softserve/course_evaluation/controller/QuestionController.java [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+@RequestMapping("api/v1/questions")
+public class QuestionController {
+
+    private final QuestionService questionService;
+
+    public QuestionController(QuestionService questionService) {
+        this.questionService = questionService;
+    }
+
+    @GetMapping("")
```
[the remark this answers is not in the document]
Comment (FedenkoM) [reply src/main/java/ita/softserve/course_evaluation/controller/QuestionController.java]: Remove "" from get method

File: src/main/java/ita/softserve/course_evaluation/controller/QuestionController.java [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    public ResponseEntity<List<QuestionDto>> getAllQuestions(){
+        final List<QuestionDto> questions = questionService.getAllQuestion();
+
+        return questions != null && !questions.isEmpty()
+                ? new ResponseEntity<>(questions, HttpStatus.OK)
+                : new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+
+    @PostMapping("/addQuestion")
```
[the remark this answers is not in the document]
Comment (FedenkoM) [reply src/main/java/ita/softserve/course_evaluation/controller/QuestionController.java]: edited

File: src/main/java/ita/softserve/course_evaluation/controller/QuestionController.java [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+        return new ResponseEntity<>(questionService.findQuestionById(questionId), HttpStatus.OK);
+//        Question question = questionService.getQuestionById(questionId);
+//        return question != null
+//                ? new ResponseEntity<>(question, HttpStatus.OK)
+//                : new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @PutMapping("{id}")
+    public ResponseEntity<QuestionDto> updateEmployee(@PathVariable("id") long id
+            ,@RequestBody QuestionDto question){
```
[the remark this answers is not in the document]
Comment (FedenkoM) [reply src/main/java/ita/softserve/course_evaluation/controller/QuestionController.java]: done

File: src/main/java/ita/softserve/course_evaluation/service/AnswerToFeedbackService.java (hunk 1 above)
Comment (FedenkoM) [reply src/main/java/ita/softserve/course_evaluation/service/AnswerToFeedbackService.java:15]: added blank line

File: src/main/java/ita/softserve/course_evaluation/entity/Feedback.java (hunk 4 above)
Comment (FedenkoM) [reply src/main/java/ita/softserve/course_evaluation/entity/Feedback.java:16]: done

File: src/main/java/ita/softserve/course_evaluation/service/QuestionService.java (hunk 2 above)
Comment (FedenkoM) [reply src/main/java/ita/softserve/course_evaluation/service/QuestionService.java:17]: done

File: src/main/java/ita/softserve/course_evaluation/controller/QuestionController.java [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+        return new ResponseEntity<>(questionService.saveQuestion(question),
+                HttpStatus.CREATED);
+    }
+
+    @GetMapping("{id}")
+    public ResponseEntity<QuestionDto> getQuestionById(@PathVariable("id") long questionId){
+        return new ResponseEntity<>(questionService.findQuestionById(questionId), HttpStatus.OK);
+//        Question question = questionService.getQuestionById(questionId);
+//        return question != null
+//                ? new ResponseEntity<>(question, HttpStatus.OK)
```
[the remark this answers is not in the document]
Comment (FedenkoM) [reply src/main/java/ita/softserve/course_evaluation/controller/QuestionController.java]: removed

**Review by Forfend — APPROVED** [review]

**Pull request merged**